### PR TITLE
chore(Makefile): Use immutable tags

### LINKED ISSUE #64: Publish immutable artifact tags + canary version to Docker registries
We want to have the ability to have deis/charts reference a specific immutable image which will enable us better traceability on issues that come into various components.

What we've done with efforts in other repositories is publish a version:
- `git-[short sha]` that will never be overwritten
- `canary` that will move to reference the latest version available of a component

Examples of the easiest way to roll this change out can be seen in deis/workflow#487, deis/builder#246, and deis/workflow-manager#11

### PR DESCRIPTION
**Pull request by krancour**
Fixes #64

### CODE REVIEW

**Comment by mboersma** [comment]
This failed for reasons I'm having a hard time correlating to the PR. Maybe we found flakiness in the backup tests?

```
-----> checking if database is running
pg_ctl: server is running (PID: 1)
/usr/lib/postgresql/9.4/bin/postgres
-----> checking if minio has 5 backups
!!!    did not find 5 base backups, which is the default (found 6)
make: *** [test] Error 1
```

File: Makefile [hunk 1]
```diff
@@ -3,13 +3,11 @@
...
-VERSION ?= git-$(shell git rev-parse --short HEAD)
-
-# Legacy support for DEV_REGISTRY, plus new support for DEIS_REGISTRY.
 DEIS_REGISTY ?= ${DEV_REGISTRY}/
 IMAGE_PREFIX ?= deis
 
+include versioning.mk
+
 # Canonical docker image name
 IMAGE := ${DEIS_REGISTRY}${IMAGE_PREFIX}/${SHORT_NAME}:${VERSION}
```
Comment (bacongobbler) [inline Makefile]: should this be removed since it's defined in versioning.mk?

**Comment by bacongobbler** [comment]
@mboersma indeed I think you did. I'll get back to this in a sec and see what's going on.

File: Makefile (hunk 1 above)
Comment (krancour) [inline Makefile]: Yes.  My bad.  Fixing...

**Comment by bacongobbler** [comment]
LGTM. I'll check out the flakiness in a different issue.

**Comment by bacongobbler** [comment]
@krancour you're good to :shipit:

**Pull request merged**